README.md

```diff
@@ -25,39 +40,130 @@ False
 ```
 It's that easy.
 
-You can create a dark mode switch listener daemon thread with `darkdetect.listener` and pass a callback function. The function will be called with string "Dark" or "Light" when the OS switches the dark mode setting.
+### Listener
+
+`darkdetect` exposes a listener API which is far more efficient than busy waiting on `theme()` changes.
+This API is exposed primarily via a `Listener` class.
+The `darkdetect.Listener` class exposes the following methods / members:
+
+##### `.__init__(callback: Optional[Callable[[str], None]])`
+
+The construct simply sets `.callback` to the given callback argument
+
+##### `.callback: Optional[Callable[[str], None]]`
+
+The callback function that the listener uses.
+The function will be called with string "Dark" or "Light" when the OS
+It is safe to change this during program execution.
+It is safe to set this value to `None`, while the listener will still be active,
+theme changes will not invoke the callback; though running callbacks will not be interrupted.
+This is useful if 'temporarily pausing' the listener is desired.
+
+##### `.listen()`
+
+This starts listening for theme changes, it will invoke
+`self.callback(theme_name)` when a change is detected.
+
+After a listener is stopped successfully via `.stop` (the return value must be `True`),
+it can be started again via `.listen()`.
+New listeners may be constructed, should waiting for `.stop` not be desired.
+
+##### `.stop(timeout: Optional[int]) -> bool`
+
+This function initiates the listener stop sequence and
+waits for the listener to stop for at most `timeout` seconds.
+This function returns `True` if the listener successfully
+stops before the timeout expires; otherwise `False`.
+`timeout` may be any non-negative integer or `None`.
+After `.stop` returns, regardless of the argument passed to it,
+theme changes will no longer invoke the callback
+Running callbacks will not be interrupted and may continue executing.
 
-``` python
+`.stop` may safely be re-invoked any number of times.
+Calling `.stop(None)` after `.stop(0)` will work as expected.
+
+In most cases `.stop(0)` should be sufficient as this will successfully
+prevent future theme changes from generating callbacks.
+The two primary use cases for `stop` with a timeout are:
+1. Cleaning up listener resources (be that subprocesses or something else)
+2. To restart the existing listener; a listener's `.listen()` function
+may only be re-invoked if a call to `.stop` has returned `True`.
```

Comment:
I can do this if it's preferred; just let me know what exactly you are thinking. Personally I think it is ok for README's to be not-super short. In my opinion, even with this API, this project's documentation is fairly small; you could remove some of the text from this README like these lines if character count is the concern without issue, I just figured I'd try to provide an abundance of information to avoid any confusion or undefined behavior or anything.

In my opinion, `/doc` might be less preferred to either a single README.md or a readthedocs (though if you prefer the latter I would ask you to do instead of me as I don't know how nor have I use rst / sphinx extensively).

Reply:
I would prefer a slim README and everything else in a `/docs` folder with Markdown files. I let you decide how to structure the further files. You may link the files in `/docs` from the README, if you want.